### ISSUE TITLE
Add new component specification: Data Grid

### ISSUE THREAD

**Issue by chrisdholt** [microsoft/fast#3750]
A new specification needs to be created for "Data Grid" as a foundational component. Data Grids are commonly requested controls and used in different variations across numerous design systems.

This issue should be closed with the introduction of the Data Grid specification, and a new issue should be opened to create the control once the spec is complete.
